includes/snmp.inc.php

```diff
@@ -37,22 +37,63 @@ function prep_snmp_setting($device, $setting)
     }
 }//end prep_snmp_setting()
 
+/**
+ * @param $device
+ * @return array $extra will contain a list of mib dirs
+ */
+function get_mib_dir($device)
+{
+    global $config;
+    $extra = array();
+
+    if (file_exists($config['mib_dir'] . '/' . $device['os'])) {
+        $extra[] = $config['mib_dir'] . '/' . $device['os'];
+    }
+
+    if (file_exists($config['mib_dir'] . '/' . $device['os_group'])) {
+        $extra[] = $config['mib_dir'] . '/' . $device['os_group'];
+    }
+
+    if (isset($config['os_groups'][$device['os_group']]['mib_dir'])) {
+        if (is_array($config['os_groups'][$device['os_group']]['mib_dir'])) {
+            foreach ($config['os_groups'][$device['os_group']]['mib_dir'] as $k => $dir) {
+                $extra[] = $config['mib_dir'] . '/' . $dir;
+            }
+        }
+    }
+
+    if (isset($config['os'][$device['os']]['mib_dir'])) {
```

Comment:
I've left it so that calls with custom dirs in will continue to work as this PR is just the initial inception rather than having gone through everything, I'd do further PRs off the back of it and work through the OS'.

The idea would be that the custom dir calls are removed longer term so we abstract mib directory lookups from the user / snmp calls.